### PR TITLE
Step2

### PR DESCRIPTION
**Pull request by BogdanPerekhrest34**

### CODE REVIEW

File: app/models/authors_book.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class AuthorsBook < ApplicationRecord
+  belongs_to :author
+  belongs_to :book, dependent: :destroy
```
Comment (nico-saj) [inline app/models/authors_book.rb:5]: `belongs_to` can't have `dependent` option

File: app/services/category_service/generate_categories.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module CategoryService
+  class GenerateCategories < ApplicationService
+    def call
+      generate_categories
```
Comment (nico-saj) [inline app/services/category_service/generate_categories.rb:6]: Don't create method with only call of another method. Move logic from `generate_categories` here

File: app/services/category_service/generate_categories.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module CategoryService
+  class GenerateCategories < ApplicationService
```
Comment (nico-saj) [inline app/services/category_service/generate_categories.rb:4]: Name classes with nouns: `CategoryGenerator`

File: app/controllers/books_controller.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class BooksController < ApplicationController
```
Comment (nico-saj) [inline app/controllers/books_controller.rb:3]: U also need to add authorisation with Pundit or CanCanCan

File: db/migrate/20220217165443_create_books.rb [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  def change
+    create_table :books do |t|
+      t.string :name
+      t.float :price
+      t.text :short_description
+      t.text :full_description
+      t.string :image_data
+      t.timestamps
+    end
+    add_reference :books, :category, index: true, foreign_key: true
```
Comment (nico-saj) [inline db/migrate/20220217165443_create_books.rb:13]: `foreign_key: { on_delete: :delete_all }`

All `on_delete` options here have to math `dependent` options in rails models

File: app/views/books/_categories.html.haml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+%ul.list-inline.pt-10.mb-25.mr-240
+  %li.mr-35
+    = link_to books_path, class:'filter-link' do
+      All
```
Comment (nico-saj) [inline app/views/books/_categories.html.haml:4]: Move text to I18n

**Review by nico-saj — CHANGES_REQUESTED** [review]

File: db/migrate/20220314154821_update_books.rb [hunk 7]
```diff
@@ -0,0 +1,6 @@
+class UpdateBooks < ActiveRecord::Migration[6.1]
```
Comment (nico-saj) [inline db/migrate/20220314154821_update_books.rb:1]: U don't need to create new migrations into single PR. It's ok to rollback several migrations, update, and run ones

File: db/migrate/20220217213055_create_table_authors_books.rb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class CreateTableAuthorsBooks < ActiveRecord::Migration[6.1]
+  def change
+    create_table :authors_books do |t|
+      t.belongs_to :author, index: true
```
Comment (nico-saj) [inline db/migrate/20220217213055_create_table_authors_books.rb:6]: here we also need to add `foreign_key: { on_delete: ...}`

File: app/views/devise/mailer/confirmation_instructions.html.erb [hunk 9]
```diff
@@ -0,0 +1,5 @@
+<p>Welcome <%= @email %>!</p>
```
Comment (nico-saj) [inline app/views/devise/mailer/confirmation_instructions.html.erb:1]: I18n and check other files

**Review by nico-saj — CHANGES_REQUESTED** [review]

File: app/views/devise/mailer/confirmation_instructions.html.erb (hunk 9 above)
Comment (BogdanPerekhrest34) [reply app/views/devise/mailer/confirmation_instructions.html.erb]: Files which belong to this step I fixed but devise files belongs to another step and in another step I'll fix this

File: db/migrate/20220318154331_update_authors_books.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+class UpdateAuthorsBooks < ActiveRecord::Migration[6.1]
```
Comment (nico-saj) [inline db/migrate/20220318154331_update_authors_books.rb:1]: U mustn't create new migration files within one PR. Just rollback migrations, update files and run updated migrations.

**Review by nico-saj — CHANGES_REQUESTED** [review]

File: app/controllers/books_controller.rb [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  load_and_authorize_resource
+  load_and_authorize_resource :category
+  include Pagy::Backend
+
+  def index
+    @pagy, @books, @categories = Books::BooksGenerator.new(params, books: @books, categories: @categories).call
+  end
+
+  def create
+    if Book.create(book_params)
```
Comment (nico-saj) [inline app/controllers/books_controller.rb:13]:
```
@book.assign_attributes(book_params)
if @book.save
...
```

**Review by nico-saj — CHANGES_REQUESTED** [review]

File: db/migrate/20220218081149_add_columns_to_books.rb [hunk 12]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
```
Comment (nico-saj) [inline db/migrate/20220218081149_add_columns_to_books.rb:1]: The same here. U don't need to update table here. rollback, update, and run migrations

**Review by nico-saj — APPROVED** [review]

**Pull request merged**